docs/usage/app-state.rst

```diff
@@ -0,0 +1,157 @@
+=================
+Application State
+=================
+
+Its possible to set & manipulate the Starlite application's `state` using a couple of
+`application hooks`. We discussed more about these application hooks & the techniques to
+use them in a later section of the documentations.
+
+Hence for now, know that the main advantage of using the application ``state`` is its
+accessibility during multiple stages of the connections. On top of it, the same ``state``
+can be injected in to dependencies & route handlers. Since the said `application state` is
+an instance & is accessible via the ``app.state`` attribute, as such it can be accessed
+wherever the app instance is accessible.
+
+Its important to understand in this context that the application instance is injected
+into the ASGI ``scope`` mapping for each connection (i.e request or websocket connection)
+as ``scope[app].state``. This makes the application accessible whenever the scope mapping
+is available, e.g in middleware, on |Request|_ & |Websocket|_ instances (accessible as
+``request.app`` & ``socket.app`` respectively) as well as many other places.
+
+Therefore, `state` offers an easy way to share contextual data between disparate parts of
+the application as described below:
+
+.. code-block:: python
+
```

Comment:
Please use `literalinclude`. We don't want inline code, since all examples should be tested, which is easier to do if they're just `.py` files.

Reply:
Yeah makes sense. I'll move all the example code snippets to a separate folder named `examples` then.

Reply:
It already exists. https://github.com/starlite-api/starlite/tree/main/examples. What's not in there can be extracted. Take a look at [this section](https://github.com/starlite-api/starlite/blob/main/CONTRIBUTING.md#cleaning-up-examples) in `CONTRIBUTING.md`.